13743 v box is not working in 3.4.0rc1

GillesFa opened the pull request:
- [x] issues: fixes #13743
- [x] tests added / passed

GillesFa committed:
Fixed the Hbox and VBox

GillesFa committed:
created a dummy commit to trigger the pre-commit hooks

GillesFa committed:
undo previous commit

bryevdv commented on tests/unit/bokeh/models/test_layouts__models.py:
```diff
@@ -70,6 +70,25 @@ def test_Column() -> None:
     check_children_prop(Column)
 
+def check_children_prop_boxes(layout_callable: type[HBox | VBox]):
+    ## component subclasses are layouts, widgets and plots
```
nit: 
```suggestion
    # component subclasses are layouts, widgets and plots
```

bryevdv commented on tests/unit/bokeh/models/test_layouts__models.py:
```diff
@@ -70,6 +70,25 @@ def test_Column() -> None:
+    layout2 = layout_callable(children=components)
+    assert layout2.children == [{'child': c} for c in components]
+    assert layout2._check_repeated_layout_children() == []
+    # Test value error raised when non-layout is provided as children
```
It would be ideal to also add a duplicate after this, and then verify that the check finds the duplicate

bryevdv commented:
Hi @GillesFa Thanks for the PR! Just two small comments to consider. You don't need to worry about the BokehJS failures, they are unrelated. We are currently working through some instability in our CI checks.

GillesFa committed:
Addressed minor comments in PR

GillesFa commented:
Hi @bryevdv I just addressed your comments above.
Please could you guys check the EOL are correct. I think they are correct, but with all the fiddling I had to do, I would prefer that someone check it too!

mattpap commented:
@GillesFa, thanks.

bryevdv commented:
Thanks @GillesFa !
